src/parser/parse.ts

```diff
@@ -0,0 +1,204 @@
+            value: { name: packageName.identifier, newPos: pos + 2 },
+        }
+    } else {
+        return {
+            resultType: 'error',
+            message: 'the ipkg file needs to start with a package description',
+        }
+    }
+}
+
+/**
+ * Return the proper field value from a token.
+ * Return `null` if it's not a valid token.
+ */
+function tokenValueToFieldValue(
+    token: UsefulToken,
+): TextField | IdentifierField | null {
+    switch (token.tokenType) {
+        case 'ident': {
+            return { fieldType: 'identifier', identifier: token.identifier }
+        }
+        case 'string-lit': {
+            return { fieldType: 'text', text: token.string }
+        }
+        default: {
+            return null
+        }
+    }
+}
+
+/**
+ * Parse a field value. The thing(s) after the "=".
+ */
+function parseFieldValue(
+    tokens: Array<UsefulToken>,
+    pos: number,
+): ParseResult<{ fieldValue: FieldValue; newPos: number }> {
+    let values = [tokens[pos]]
+    let newPos = pos + 1
+    while (
+        tokens[newPos] &&
+        tokens[newPos].tokenType === 'separator' &&
+        tokens[newPos + 1]
+    ) {
+        values.push(tokens[newPos + 1])
+        newPos += 2
+    }
+
+    // convert the parsed values to the correct `FieldValue`.
+    if (values.length === 0) {
+        return {
+            resultType: 'error',
+            message: 'expected at least one value after "="',
+        }
+    } else if (values.length === 1) {
+        const fieldValue = tokenValueToFieldValue(values[0])
+        if (fieldValue !== null) {
+            return {
+                resultType: 'success',
+                value: {
+                    fieldValue,
+                    newPos: newPos,
+                },
+            }
+        } else {
+            return {
+                resultType: 'error',
+                message: `invalid value after the "=": ${tokensToString(
+                    values,
+                )}`,
+            }
+        }
+    } else {
+        const fieldValues: Array<TextField | IdentifierField> = []
+        for (const token of values) {
+            const value = tokenValueToFieldValue(token)
+            if (value !== null) {
+                fieldValues.push(value)
+            }
+        }
+        if (fieldValues.length === values.length) {
+            return {
+                resultType: 'success',
+                value: {
+                    fieldValue: {
+                        fieldType: 'list',
+                        list: fieldValues,
+                    },
+                    newPos: newPos,
+                },
+            }
+        } else {
+            return {
+                resultType: 'error',
+                message: `invalid values after the "=": ${tokensToString(
+                    values,
+                )}`,
+            }
+        }
+    }
+}
+
+/**
+ * Parse a field of the iPKG file.
+ *
+ * E.g: `main = Idris.Main`
```

Comment:
Maybe also show the result, so that the reader knows that the dot in "Idris.Main" is not special and the whole thing is just one token/value.